Utilities/ocvalidate/ValidateUEFI.c

```diff
@@ -0,0 +1,279 @@
+  BOOLEAN                   IsPointerSupportEnabled;
+  CONST CHAR8               *Resolution;
+  UINT32                    UserWidth;
+  UINT32                    UserHeight;
+  UINT32                    UserBpp;
+  BOOLEAN                   UserSetMax;
+  CONST CHAR8               *AsciiAudioDevicePath;
+  EFI_DEVICE_PATH_PROTOCOL  *AudioDevicePath;
+  CHAR16                    *UnicodeAudioDevicePath;
+  CHAR16                    *TextualAudioDevicePath;
+
+  DEBUG ((DEBUG_VERBOSE, "config loaded into UEFI checker!\n"));
+
+  ErrorCount                       = 0;
+  IndexOpenUsbKbDxeEfiDriver       = 0;
+  IndexPs2KeyboardDxeEfiDriver     = 0;
+  UserUefi                         = &Config->Uefi;
+  UserMisc                         = &Config->Misc;
+  HasOpenRuntimeEfiDriver          = FALSE;
+  HasOpenUsbKbDxeEfiDriver         = FALSE;
+  HasPs2KeyboardDxeEfiDriver       = FALSE;
+  IsRequestBootVarRoutingEnabled   = UserUefi->Quirks.RequestBootVarRouting;
+  IsKeySupportEnabled              = UserUefi->Input.KeySupport;
+  IsPointerSupportEnabled          = UserUefi->Input.PointerSupport;
+  PointerSupportMode               = OC_BLOB_GET (&UserUefi->Input.PointerSupportMode);
+  KeySupportMode                   = OC_BLOB_GET (&UserUefi->Input.KeySupportMode);
+  IsClearScreenOnModeSwitchEnabled = UserUefi->Output.ClearScreenOnModeSwitch;
+  IsIgnoreTextInGraphicsEnabled    = UserUefi->Output.IgnoreTextInGraphics;
+  IsReplaceTabWithSpaceEnabled     = UserUefi->Output.ReplaceTabWithSpace;
+  IsSanitiseClearScreenEnabled     = UserUefi->Output.SanitiseClearScreen;
+  TextRenderer                     = OC_BLOB_GET (&UserUefi->Output.TextRenderer);
+  IsTextRendererSystem             = FALSE;
+  ConsoleMode                      = OC_BLOB_GET (&UserUefi->Output.ConsoleMode);
+  Resolution                       = OC_BLOB_GET (&UserUefi->Output.Resolution);
+  AsciiAudioDevicePath             = OC_BLOB_GET (&UserUefi->Audio.AudioDevice);
+  AudioDevicePath                  = NULL;
+  UnicodeAudioDevicePath           = NULL;
+  TextualAudioDevicePath           = NULL;
+
+  //
+  // Sanitise strings.
+  //
+  if (AsciiStrCmp (TextRenderer, "BuiltinGraphics") != 0
+    && AsciiStrCmp (TextRenderer, "BuiltinText") != 0
+    && AsciiStrCmp (TextRenderer, "SystemGraphics") != 0
+    && AsciiStrCmp (TextRenderer, "SystemText") != 0
+    && AsciiStrCmp (TextRenderer, "SystemGeneric") != 0) {
+    DEBUG ((DEBUG_WARN, "UEFI->Output->TextRenderer is illegal (Can only be BuiltinGraphics, BuiltinText, SystemGraphics, SystemText, or SystemGeneric)!\n"));
+    ++ErrorCount;
+  } else {
+    //
+    // Check whether TextRenderer has System prefix.
+    //
+    if (AsciiStrnCmp (TextRenderer, "System", L_STR_LEN ("System")) == 0) {
+      IsTextRendererSystem = TRUE;
+    }
+  }
+
+  if (UserUefi->Apfs.EnableJumpstart
+    && (UserMisc->Security.ScanPolicy != 0 && (UserMisc->Security.ScanPolicy & OC_SCAN_ALLOW_FS_APFS) == 0)) {
```

Comment:
This check is not correct. First part should be `(UserMisc->Security.ScanPolicy & OC_SCAN_FILE_SYSTEM_LOCK) != 0`. I.e. the condition should be "FS restrictions enabled and APFS FS disabled"

Reply:
https://github.com/PMheart/OpenCorePkg/commit/bebff302b3b15c23f9fcf4f7ba101fa8326cc632